src/providers/postgres/raster/qgspostgresrasterprovider.cpp

```diff
@@ -2210,7 +2210,7 @@ QString QgsPostgresRasterProvider::pkSql()
   if ( mPrimaryKeyAttrs.count( ) > 1 )
   {
     QStringList pkeys;
-    for ( const auto &k : std::as_const( mPrimaryKeyAttrs ) )
+    for ( const QString &k : std::as_const( mPrimaryKeyAttrs ) )
```

Comment:
explicit is better than implicit, says the zen of a language with duck typing

Reply:
Totally agree -- this is the inherent danger that I see in auto... if we decided to change mPrimaryKeyAttrs to QList< int > then who knows what the loop would do as a result :scream: